GVFS/GVFS.Common/NamedPipes/NamedPipeMessages.cs

```diff
@@ -137,6 +138,46 @@ public Message CreateMessage()
             }
         }
 
+        public static class PostIndexChanged
+        {
+            public const string NotificationRequest = "PICN";
+            public const string SuccessResult = "S";
+            public const string FailureResult = "F";
+
+            public class Request
+            {
+                public Request(Message message)
+                {
+                    if (message.Body.Length != 2)
+                    {
+                        throw new InvalidOperationException($"Invalid PostIndexChanged message. Expected at least 2 characters, got: {message.Body.Length} from message: '{message.Body}'");
+                    }
+
+                    this.WorkingDirectoryUpdated = message.Body[0] == '1';
+                    this.OnlyIndexUpdated = message.Body[1] == '1';
+                }
+
+                public bool WorkingDirectoryUpdated { get; }
+
+                public bool OnlyIndexUpdated { get; }

```

Comment:
Related to the comment in `GVFS/GVFS.PostIndexChangedHook/main.cpp` it's not clear to me what `OnlyIndexUpdated` means.  Consider adding a comment here (and\or updating the name)

Reply:
I'm going to change this to SkipWorktreeBitsUpdated.  Hopefully that makes more sense.